### DISCUSSION TITLE
<flux:sidebar> in one file, best practice?

### DISCUSSION THREAD

**Discussion by madeskpri** [livewire/flux#756] [Feature Requests] [answered]
Hi!
I was asked by Flux UI Support to post my question here so please excuse my noobie question. ;-) I'm pretty new to laravel/livewire and I would like to have the <flux:sidebar> component in one component file, so I can use the sidebar in every route. However, I am struggling with the implementation, if I put the

< flux:sidebar... >
[...]
< /flux:sidebar >

part in an individual component file and follow the "secondary header" example on https://fluxui.dev/layouts/sidebar and try to add this sidebar as < livewire:sidebar > in my routes, I get the error message that only one root element is allowed. If I use a < div > to get that, the layout obviously is no longer working. As I said, I'm sorry if this is a noobie question for you, can you please guide me to the right direction/docs/links what is the best practice to have the complete sidebar in one file and reuse it in the routes? 
Thanks! Mark

**Comment by EriksonEnterprises** [comment]
I believe you'll put the component in:

`resources/views/components/sidebar.blade.php`

(make the folder if it doesn't exist)

Put everything to do with the sidebar in there without any extra divs

Now wherever you want the sidebar, you do:

`<x-sidebar />`

You can also pass variables to it like:

`<x-sidebar currentPage='home' />`

If I remember correctly lol.

Is that what you're talking about?

**Reply by madeskpri** [reply]
Thank you! So easy, sorry for asking...

**Reply by EriksonEnterprises** [reply]
Nah you're good :) we all learn by asking!

Glad it worked :)

**Comment by jeffchown** [comment, marked as answer]
@madeskpri Sounds like a Laravel layout could be the answer to your question.
Have a look at: https://laravel.com/docs/11.x/blade#building-layouts

You could put your `flux:sidebar` in a layout that could then be used for every blade file/view

**Reply by madeskpri** [reply]
Thank you! That did the trick!

**Reply by jeffchown** [reply]
You're welcome @madeskpri
